### DISCUSSION TITLE
How to optimise dnn-mmod_face_detector?

### DISCUSSION THREAD

**Discussion by facehawk** [davisking/dlib#2850] [Q&A]
The programme failed the template creation time test, which was 2.25 seconds longer than the NIST acceptable standard of 1.5 seconds. In this sense, we did internal testing by implementing a timing check. During internal testing, it was discovered that some images took 7 seconds to analyse.

Given the above, how can we optimize the face detector or suggest an alternative to the face detector?

**Comment by arrufat** [comment]
If you have time or memory constraints, you should resize the images to have at most a certain number of pixels (the amount that's acceptable by your application).

I am assuming that you're already running the detector on GPU.

**Reply by facehawk** [reply]
We have no memory constraints. However, it is the time, which we need to meet especially CreateFaceTemplate Face Image List. 

Yes, we are internally testing it on GPU. It is giving us multi-threading/ multi-processing techniques warning during the 1:1 validation. 

How to address these issues?

**Reply by arrufat** [reply]
The inference time is proportional to the number of pixels in an image, so find out the biggest size that fits your constraints, and resize all images to have that number of pixels. That's one (probably the sinplest, too) way to do it.

**Reply by facehawk** [reply]
Thanks for your prompt response.

Now, an 8 bit gray scale image is creating problem and giving error during enrolment of multiple faces in an image. How can you address this issue?

**Reply by arrufat** [reply]
I don't understand the question.
